### DISCUSSION TITLE
Enable/disable visibility of individual filters

### DISCUSSION THREAD

**Discussion by inmanturbo** [rappasoft/laravel-livewire-tables#759] [Feature Requests]
`laravel-livewire-tables` now has support built in for hooking in filters elsewhere in the page using `events`. It is also possible to hide the defined filters so that you can use your own implementation with 
```php
    public function configure(): void
    {
         $this->setFiltersVisibilityStatus(false);
    }
```
However there isn't any way to to show some filters and not others. Say you want to add several filters, and you want to implement your own frontend for some but not the others, but you want to utilize the `laravel-livewire-tables` backend api for all of them. the proposed api is something like this:

```php
public function filters(): array
{
    return [
       // this filter will be seen in popover and slide down
            'verified_from' => DateFilter::make('Verified From')
            ->config([
                'min' => '2020-01-01',
                'max' => '2021-12-31',
            ]),
            // this filter won't  be seen in popover and slide down
           'active' =>  SelectFilter::make('Active')
            ->options([
                '' => 'All',
                'yes' => 'Yes',
                'no' => 'No',
            ])
            ->hidden() // hide the filter from popover and slidedown
    ];
}
```

This way developers can also do things like put the filters into the secondary header for some filters but not others ; -)

```php
public function columns(): array
{
    return [
        Column::make('Active')
          ->secondaryHeader(
            function () {
              return view('livewire-tables::components.tools.filters.select', [
                'component' => $this,
                 'filter' => $this->filters()['active'],
               ]);
            }
         );
    ];
}
```
@rappasoft This feature should not be too difficult to implement and I don't mind doing the work if you're not opposed to it. It's just a single method that sets a boolean property, and a conditional in the view to check for that property, perhaps a couple checkers to make it more fluent like `isVisible()` and `isHidden()`

**Comment by rappasoft** [comment]
Yeah, I'm definitely for this if you want to implement and throw on the develop branch. I just ask:

1. Update the changelog where necessary
2. Test as best as you can
3. Try to organize the way I do (or I'll move stuff before it releases)
4. Add to the docs if necessary

I've been working on figuring out how to make the query builder more flexible but I keep running into road blocks because I need to keep it backward compatible. I've been releasing small other improvements as I need them on other projects as well.

I also just bought livewire-tables.com so I can get an actual site up with more examples and code (might take a while).

Edit: This project is getting a few thousand downloads a day now I think, are we the most popular livewire tables plugin now?

**Reply by inmanturbo** [reply]
@rappasoft rather than push directly to develop, I opened a PR just so the proposed changes can be reviewed and discussed all in one place if needed. https://github.com/rappasoft/laravel-livewire-tables/pull/761
